pyerrors/input/json.py

```diff
@@ -0,0 +1,332 @@
+                retd['deltas'].append([di[1:] for di in rep['deltas']])
+                retd['is_merged'][rep['name']] = rep.get('is_merged', False)
+        retd['deltas'] = np.array(retd['deltas'])
+        return retd
+
+    def get_Obs_from_dict(o):
+        layouts = o.get('layout', '1').strip()
+        if layouts != '1':
+            raise Exception("layout is %s has to be 1 for type Obs." % (layouts), RuntimeWarning)
+
+        values = o['value']
+        od = _gen_obsd_from_datad(o['data'])
+
+        ret = Obs([[ddi[0] + values[0] for ddi in di] for di in od['deltas']], od['names'], idl=od['idl'])
+        ret.reweighted = o.get('reweighted', False)
+        ret.is_merged = od['is_merged']
+        ret.tag = o.get('tag', None)
+        return ret
+
+    def get_List_from_dict(o):
+        layouts = o.get('layout', '1').strip()
+        layout = int(layouts)
+        values = o['value']
+        od = _gen_obsd_from_datad(o['data'])
+
+        ret = []
+        for i in range(layout):
+            ret.append(Obs([list(di[:, i] + values[i]) for di in od['deltas']], od['names'], idl=od['idl']))
+            ret[-1].reweighted = o.get('reweighted', False)
+            ret[-1].is_merged = od['is_merged']
+            ret[-1].tag = o.get('tag', None)
+        return ret
+
+    def get_Array_from_dict(o):
+        layouts = o.get('layout', '1').strip()
+        layout = [int(ls.strip()) for ls in layouts.split(',') if len(ls) > 0]
+        values = o['value']
+        od = _gen_obsd_from_datad(o['data'])
+
+        ret = []
+        for i in range(np.prod(layout)):
+            ret.append(Obs([di[:, i] + values[i] for di in od['deltas']], od['names'], idl=od['idl']))
+            ret[-1].reweighted = o.get('reweighted', False)
+            ret[-1].is_merged = od['is_merged']
+            ret[-1].tag = o.get('tag', None)
+        return np.reshape(ret, layout)
+
+    if not fname.endswith('.json') and not fname.endswith('.gz'):
+        fname += '.json'
+    if gz:
+        if not fname.endswith('.gz'):
+            fname += '.gz'
+        with gzip.open(fname, 'r') as fin:
+            d = json.loads(fin.read().decode('utf-8'))
+    else:
+        if fname.endswith('.gz'):
+            warnings.warn("Trying to read from %s without unzipping!" % fname, UserWarning)
+        with open(fname, 'r') as fin:
+            d = json.loads(fin.read())
+
+    prog = d.get('program', '')
+    version = d.get('version', '')
+    who = d.get('who', '')
+    date = d.get('date', '')
+    host = d.get('host', '')
+    if prog and verbose:
+        print('Data has been written using %s.' % (prog))
+    if version and verbose:
+        print('Format version %s' % (version))
+    if np.any([who, date, host] and verbose):
+        print('Written by %s on %s on host %s' % (who, date, host))
+    description = d.get('description', '')
+    if description and verbose:
+        print()
+        print(description)
```

Comment:
```suggestion
        print("Description: ", description)
```
Maybe make clear that this is the file description?